### ISSUE TITLE
Implicit examples trying to set box vectors, poorly

### ISSUE THREAD

**Issue by Lnaden** [choderalab/yank#469]
The `p-xylene-implicit` example crashes on the 3rd iteration because of the OpenMM error: "Exception: First periodic box vector must be parallel to x." The traceback for this error occurs in the `repex.py` ['_propagate_replica' line](https://github.com/choderalab/yank/blob/e73ea0beabbac07f947c5eacfa15a832be2ee7fa/Yank/repex.py#L1302). There are of course 2 problems here:

1. Implicit solvents simulations shouldn't be trying to set box vectors in the first place.
2. Something is wrong with the Barostat. Its either:
 * Its incorrectly setting the rectangular box vectors so they do not look like 
 ```python
[[ X, 0, 0],
 [ 0, Y, 0],
 [ 0, 0, Z]]
```
 * Its incorrectly setting the triclinic box so they do not look like (if I remember the OpenMM rules for triclinic boxes) 
 ```python
[[ X, 0, 0],
 [ Y, Y, 0],
 [ Z, Z, Z]]
```
 * Its actually throwing NaN's and those are being passed before the built in sanity checks.

Not sure yet if this is relegated to only this example or if the problem is more widespread.

**Comment by Lnaden** [comment]
I have narrowed the problem down to something crashing in the actual `step` function of `propagate_replica`. Before the crash, all the `replica_box_vectors` meet the 
``` 
[[ X, 0, 0],
[ 0, Y, 0],
[ 0, 0, Z]]
```
formatting, then at some point in the `step` function, something goes haywire. Again, this problem may be solved by first figuring out *why* the implicit solvent is updating its periodic boxes in the first place.

**Comment by andrrizzi** [comment]
I've dug into this a little.

1) The box vectors are set by default by the OpenMM parser even if implicit solvent is used. Not sure if this is a bug or expected. A minimal working example:
```python
import os
from simtk import openmm
from yank import utils

abl_imatinib_dir = utils.get_data_filename(os.path.join('..', 'examples', 'abl-imatinib-implicit', 'setup'))
prmtop_path = os.path.join(abl_imatinib_dir, 'complex.prmtop')
inpcrd_path = os.path.join(abl_imatinib_dir, 'complex.inpcrd')

topology_file = openmm.app.AmberPrmtopFile(prmtop_path)
box_vectors = openmm.app.AmberInpcrdFile(inpcrd_path).boxVectors
system = topology_file.createSystem(nonbondedMethod=openmm.app.NoCutoff, implicitSolvent=openmm.app.OBC2)

>>> print system.getDefaultPeriodicBoxVectors()
[Quantity(value=(2.0, 0.0, 0.0), unit=nanometer), Quantity(value=(0.0, 2.0, 0.0), unit=nanometer), Quantity(value=(0.0, 0.0, 2.0), unit=nanometer)]
>>> print box_vectors
None
```

2) I'm not sure about this but John recently added [the code that add the barostat even in vacuum](https://github.com/choderalab/yank/blob/master/Yank/yank.py#L334-L354). There's still [this line](https://github.com/choderalab/yank/blob/master/Yank/yank.py#L380) though that sets `pressure` to `None`. I think that's what's causing the exception. I'll test this tomorrow.

**Comment by andrrizzi** [comment]
I can confirm that the crash goes away with the removal of the code for the addition of the barostat. I'll shortly open a new issue discussing how to handle the barostat.

**Comment by andrrizzi** [comment]
Changed my mind. I'll just open a PR to implement the fix.

**Comment by jchodera** [comment]
@andrrizzi : Thanks for diving into this!

Quick comments:
* We don't want to add a barostat to implicit solvent. OpenMM should really just fail with an exception if a `Context` is created for a non-periodic system. Can you raise an OpenMM issue for that, and make sure YANK doesn't try to add a barostat to non-periodic systems?
* Deprecating our own detection of periodic boundary conditions is a great idea---OpenMM recently added a way to do that, and this method should be used.
* We don't need to set the box vectors to anything special for non-periodic systems---OpenMM should just ignore them. Setting them to `None` would cause problems, I imagine.

**Comment by andrrizzi** [comment]
This was fixed in #475 .
